pkg/client/fake/client.go

```diff
@@ -79,28 +81,42 @@ func NewFakeClientWithScheme(clientScheme *runtime.Scheme, initObjs ...runtime.O
 }
 
 func (t versionedTracker) Create(gvr schema.GroupVersionResource, obj runtime.Object, ns string) error {
-	if accessor, err := meta.Accessor(obj); err == nil {
-		if accessor.GetResourceVersion() == "" {
-			accessor.SetResourceVersion("1")
-		}
-	} else {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
 		return err
 	}
+	if accessor.GetResourceVersion() != "" {
+		return kerrors.NewBadRequest("resourceVersion can not be set for Create requests")
```

Comment:
I am not 100% sure if this is the correct error, can anybody confirm?

Reply:
Am I understanding this correctly...This check is basically doing what the API server would do?

Reply:
I found https://github.com/kubernetes/kubernetes/issues/71435 which makes me think this should be a [`NewInvalid` instead](https://godoc.org/k8s.io/apimachinery/pkg/api/errors#NewInvalid)

Reply:
Yes, exactly

Reply:
turns out I can't read https://github.com/kubernetes/kubernetes/issues/71435#issuecomment-486688084 bad request is right.

Reply:
turns out I can't read https://github.com/kubernetes/kubernetes/issues/71435#issuecomment-486688084 bad request is right.